pkg/registry/core/service/storage/rest.go

```diff
@@ -248,6 +247,9 @@ func (al *RESTAllocStuff) allocClusterIPs(service *api.Service, toAlloc map[api.
 			allocated[family] = allocatedIP.String()
 		} else {
 			parsedIP := netutils.ParseIPSloppy(ip)
+			if parsedIP == nil {
```

Comment:
We don't need that. It is parsed as part of validation: https://github.com/kubernetes/kubernetes/blob/master/pkg/apis/core/validation/validation.go#L6361

Does this gets called before Validation?

Reply:
I think we do a disservice by not preparing for bugs.  The "None" bug slipped through because we don't handle this "can't happen" case which did happen.  I'd like to see us handle more such cases as "WTF?!?" errors rather than crash or silently ignore